### PR TITLE
Rpc 0.1.0 incomplete - only test pass

### PR DESCRIPTION
**Pull request by tabaktoni**
## Motivation and Resolution
Fix breaking Tests and open release pipeline

### RPC version (if applicable)
0.1.0

## Usage-related changes
RPC Fixes related to:
- getBlock
- getCode removed
- getStorageAt
- getClassAt

## Development related changes
RPC Provider has:
- getBlockWithTxHashes
- getBlockWithTxs
- getTransactionByHash
- getTransactionByBlockIdAndIndex

getClassAt now has 2 parameters for RPC also same as sequencer (contractAddress, blockIdentifier)

openrpc.ts specify mapping of open RPC spec to TS types (work in progress)

## Checklist:
- [x] Performed a self-review of the code
- [x] Rebased to the last commit of the target branch (or merged it into my branch)
- [ ] Linked the issues which this PR resolves
- [x] Documented the changes
- [x] Updated the docs (www)
- [x] Updated the tests
- [x] All tests are passing

### CODE REVIEW

File: src/types/provider.ts [hunk 1]
```diff
@@ -1,24 +1,19 @@
...
-  old_root?: string;
   parent_hash: string;
-  sequencer: string;
   status: Status;
   transactions: Array<string>;
-  starknet_version?: string;
-}
-
-export interface GetCodeResponse {
-  bytecode: string[];
```
Comment (janek26) [inline src/types/provider.ts:20]: we heavily rely on this to detect if a contract is deployed.

I would suggest to keep it in the common interface and throw "Not implemented" error in rpc provider

File: __tests__/sequencerProvider.test.ts [hunk 2]
```diff
@@ -43,6 +46,11 @@ describeIfSequencer('SequencerProvider', () => {
       expect(transactionTrace).toHaveProperty('signature');
     });
 
+    test('getCode() -> { bytecode }', async () => {
```
[the remark this answers is not in the document]
Comment (tabaktoni) [reply __tests__/sequencerProvider.test.ts:49]: No

File: __tests__/defaultProvider.test.ts [hunk 3]
```diff
@@ -266,13 +250,13 @@ describe('defaultProvider', () => {
       });
 
       describe('Contract methods', () => {
-        let contractAddress: string;
+        // let contractAddress: string;
```
[the remark this answers is not in the document]
Comment (tabaktoni) [reply __tests__/defaultProvider.test.ts:253]: This is a temporary comment fix until I manage to complete RPC as I hardcoded provider.getClassAt() and it should get use this and block identifier

File: __tests__/defaultProvider.test.ts [hunk 4]
```diff
@@ -46,27 +46,21 @@ describe('defaultProvider', () => {
     test(`getBlock(blockHash=${exampleBlockHash}, blockNumber=undefined)`, () => {
       return expect(testProvider.getBlock(exampleBlockHash)).resolves.not.toThrow();
     });
-
```
Comment (ivpavici) [inline __tests__/defaultProvider.test.ts:49]: nitpick: new line here

File: src/types/provider.ts (hunk 1 above)
Comment (tabaktoni) [reply src/types/provider.ts:20]: Hm,
Can we create **isContractDeployed** in a common interface that would return true/false, and then RPC can implement it with function 'starknet_getClassHashAt' that uses the same parameters contact address & blockIdentifier, and sequencer can implement it with getCode ?
https://github.com/starkware-libs/starknet-specs/blob/30e5bafcda60c31b5fb4021b4f5ddcfc18d2ff7d/api/starknet_api_openrpc.json#L267

File: __tests__/sequencerProvider.test.ts [hunk 5]
```diff
@@ -7,9 +7,11 @@ import {
   getTestProvider,
 } from './fixtures';
 
+// This run only if Devnet Sequencer ?
```
Comment (ivpavici) [inline __tests__/sequencerProvider.test.ts:10]: did we figure this out? just for reference

File: src/types/provider.ts (hunk 1 above)
Comment (janek26) [reply src/types/provider.ts:20]: I would suggest to simply keep the method so dapps/wallets dont need to use a new api

RPC can throw as I said

In the mid-term this method should be dropped and both sequencer and rpc should just work with the `starknet_getClassHashAt` pattern

**Review by ivpavici — APPROVED** [review]

**Review by janek26 — APPROVED** [review]

**Pull request merged**